### PR TITLE
[package] first draft for adding jwt-cpp

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **jwt-cpp/0.3.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the  [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jwt-cpp/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            bin_path = "example"
+            if self._is_multi_configuration:
+                bin_path = os.path.join(str(self.settings.build_type), bin_path)
+            self.output.info(bin_path)
+            self.run(bin_path, run_environment=True)
```
Comment (prince-chrismc) [inline recipes/jwt-cpp/all/test_package/conanfile.py]: I copied pasted this from another project, I am not sure why the executable path is incorrect, it works on my windows system

Any help would be appreciated

Comment (madebr) [reply recipes/jwt-cpp/all/test_package/conanfile.py]: See https://github.com/prince-chrismc/conan-center-index/pull/1

File: recipes/jwt-cpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        self.requires("picojson/1.3.0")
+        self.requires("openssl/1.1.1d")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        header_dir = os.path.join(self._source_subfolder, "include", "jwt-cpp/")
```
Comment (uilianries) [inline recipes/jwt-cpp/all/conanfile.py]:
```suggestion
        header_dir = os.path.join(self._source_subfolder, "include", "jwt-cpp")
```

Don't need path separator

File: recipes/jwt-cpp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        self.copy("jwt.h", dst="include", src=header_dir, keep_path=False)
+        self.copy("base.h", dst="include", src=header_dir, keep_path=False)
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.info.header_only()
```
Comment (uilianries) [inline recipes/jwt-cpp/all/conanfile.py]:
```suggestion
```

It's not required since it doesn't contain any settings.

**Comment by gocarlos** [comment]
@uilianries @danimtb what to do with naming collisions?

E.g. https://github.com/conan-io/conan-center-index/pull/1031 asks for a conan package, this library is also called jwt-cpp -> in npm that would be `@pokowaka/jwt-cpp` and this recipe would be for `@thalhammer/jwt-cpp`

**Comment by Croydon** [comment]
There is no simple answer to this.

One aspect is simply which is added here earlier, but it is also important to look what other package managers did.

In this case I think it is okay, that `Thalhammer/jwt-cpp` claims the `jwt-cpp` name

  * we have here a PR already (not yet for the other one)
  * it has more stars/forks/watches so it might be more popular
  * vcpkg is using the `jwt-cpp`  name for `Thalhammer/jwt-cpp` as well https://github.com/microsoft/vcpkg/blob/d0a465518ef185b3c0c9ef0cfced221181654448/ports/jwt-cpp/portfile.cmake#L5

**Comment by prince-chrismc** [comment]
@uilianries I have applied the changes you requested but GitHub is not updating, if you can please take a second look and update your review that would be great =)

**Comment by prince-chrismc** [comment]
Dump! If someone can get that pending review in please.

**Comment by prince-chrismc** [comment]
@danimtb  may this please be merged?

**Comment by prince-chrismc** [comment]
Bump! Can this please be merged. It would be greatly appreciated!

**Comment by uilianries** [comment]
@prince-chrismc be a little patience, CCI is under EAP and as you can see, the CI fails sometimes, or partially upload the packages. We are putting our efforts fixing the CCI service, to provide a better experience for the community. Recipes are cool, but if the service is broken, we can't share the packages after merging a PR.

**Comment by prince-chrismc** [comment]
Thank you so very much! Dying to get my hands this! :100: 

Very much appreciated :1st_place_medal:

**Pull request merged**